gobblin-runtime/src/main/java/gobblin/runtime/util/FSJobCatalogJobSpecHelper.java

```diff
@@ -0,0 +1,157 @@
+package gobblin.runtime.util;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.URI;
+import java.util.Properties;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Optional;
+import com.typesafe.config.Config;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.runtime.api.JobSpec;
+import gobblin.runtime.api.JobSpecNotFoundException;
+import gobblin.util.ConfigUtils;
+import gobblin.util.PathUtils;
+
+
+public class FSJobCatalogJobSpecHelper {
+
+  // Key used in the metadata of JobSpec.
+  private static final String DESCRIPTION_KEY_IN_JOBSPEC = "description";
+  private static final String VERSION_KEY_IN_JOBSPEC = "version";
+
+  /**
+   * On receiving a JobSpec object, materialized it into a real file into non-volatile storage layer.
+   * @param jobConfDirPath The configuration root directory path.
+   * @param jobSpec a {@link JobSpec}
+   */
+  public static void materializeJobSpec(Path jobConfDirPath, JobSpec jobSpec)
+      throws IOException {
+    Path path = new Path(jobConfDirPath, jobSpec.getUri().toString());
```

Comment:
Please use `PathUtils.mergePaths` instead, as this will not work when the `JobSpec` uri is absolute